esrally/reporter.py

```diff
@@ -886,10 +888,22 @@ def identity(x):
             color_smaller = console.format.green
             color_neutral = console.format.neutral
 
-        if diff > 0:
-            return color_greater("+%.5f" % diff)
-        elif diff < 0:
-            return color_smaller("%.5f" % diff)
+        if as_percentage:
```

Comment:
IMHO the duplication in this method makes it harder to understand and maintain. How about we replace the remainder of this method with the following?

```suggestion
        if as_percentage:
            diff = _safe_divide(contender - baseline, baseline) * 100.0
            precision = 2
            suffix = "%"
        else:
            diff = contender - baseline
            precision = 5
            suffix = ""

        # ensures that numbers that appear as "zero" are also colored neutrally
        threshold = 10**-precision
        formatted = f"{diff:.{precision}f}{suffix}"

        if diff >= threshold:
            return color_greater(f"+{formatted}")
        elif diff <= -threshold:
            return color_smaller(formatted)
        else:
            return color_neutral(formatted)
```

Reply:
Nice, I like this a lot better. Thank you for the suggestion!